soc/renesas/smartbond/da1469x/soc.c

```diff
@@ -36,9 +37,27 @@ static uint32_t z_renesas_cache_configured;
 
 void sys_arch_reboot(int type)
 {
-	ARG_UNUSED(type);
-
-	NVIC_SystemReset();
+	if (type == SYS_REBOOT_WARM) {
+		NVIC_SystemReset();
+	} else if (type == SYS_REBOOT_COLD) {
+		if ((SYS_WDOG->WATCHDOG_REG & SYS_WDOG_WATCHDOG_REG_WDOG_VAL_NEG_Msk) == 0) {
+			/* Cannot write WATCHDOG_REG while WRITE_BUSY */
+			while ((SYS_WDOG->WATCHDOG_REG &
+				SYS_WDOG_WATCHDOG_CTRL_REG_WRITE_BUSY_Msk) != 0) {
+
+			}
+			/* Write WATCHDOG_REG */
+			SYS_WDOG->WATCHDOG_REG = BIT(SYS_WDOG_WATCHDOG_REG_WDOG_VAL_Pos);
```

Comment:
Updating the counter value should be ignored if `WATCHDOG_REG[WDOG_EN]` is not written to one, at the same time.

Reply:
![image](https://github.com/user-attachments/assets/d1525092-5e6c-457f-8b2b-700f11509f9b)
`WATCHDOG_REG[WDOG_EN]` should be zero to update its value. For sure the register field name is misleading.

Reply:
Right. Somehow, I misinterpreted the description.